src/collection-view.js

```diff
@@ -30,73 +26,54 @@ var CollectionView = Backbone.View.extend({
   //
   // option to pass `{comparator: compFunction()}` to allow the `CollectionView`
   // to use a custom sort order for the collection.
-  constructor: function(options) {
+  constructor(options) {
     this.render = _.bind(this.render, this);
 
     this._setOptions(options);
 
-    MonitorDOMRefresh(this);
-
-    this.on({
-      'before:show':   this._onBeforeShowCalled,
-      'show':          this._onShowCalled,
-      'before:attach': this._onBeforeAttachCalled,
-      'attach':        this._onAttachCalled
-    });
+    MonitorViewEvents(this);
 
     this._initBehaviors();
     this.once('render', this._initialEvents);
     this._initChildViewStorage();
-    this.initRenderBuffer();
+    this._bufferedChildren = [];
 
     Backbone.View.prototype.constructor.call(this, this.options);
 
     this.delegateEntityEvents();
   },
 
-  // Instead of inserting elements one by one into the page,
-  // it's much more performant to insert elements into a document
-  // fragment and then insert that document fragment into the page
-  initRenderBuffer: function() {
-    this._bufferedChildren = [];
-  },
-
-  startBuffering: function() {
-    this.initRenderBuffer();
-    this.isBuffering = true;
+  // Instead of inserting elements one by one into the page, it's much more performant to insert
+  // elements into a document fragment and then insert that document fragment into the page
+  _startBuffering() {
+    this._isBuffering = true;
   },
 
-  endBuffering: function() {
-    // Only trigger attach if already shown and attached, otherwise Region#show() handles this.
-    var canTriggerAttach = this._isShown && isNodeAttached(this.el);
-    var nestedViews;
+  _endBuffering() {
+    const shouldTriggerAttach = !!this._isAttached;
 
-    this.isBuffering = false;
+    this._isBuffering = false;
 
-    if (this._isShown) {
-      triggerMethodMany(this._bufferedChildren, this, 'before:show');
-    }
-    if (canTriggerAttach && this._triggerBeforeAttach) {
-      nestedViews = this._getNestedViews();
-      triggerMethodMany(nestedViews, this, 'before:attach');
+    if (shouldTriggerAttach) {
+      _.each(this._getImmediateChildren(), child => {
```

Comment:
Yes, definitely.  I don't see any public method calls between the two, so there's no sane reason the immediate children could change.